### PR TITLE
feat: add initial workspace creation

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: pkg/api/handlers/agent.go [hunk 1]
```diff
@@ -91,7 +95,20 @@ func (a *AgentHandler) Create(ctx context.Context, req api.Request) error {
 		return api.NewErrBadRequest("replace requires \"id\" in the manifest to be set")
 	}
 
-	agent := v2.Agent{
+	var (
+		agent           v2.Agent
+		createWorkspace bool
+		httpError       *api.ErrHTTP
+	)
+	if err = req.Get(&agent, spec.Manifest.ID); errors.As(err, &httpError) && httpError.Code == http.StatusNotFound {
```
Comment (thedadams) [inline pkg/api/handlers/agent.go:103]: It was easier to reason about creating the workspaces only when we need to if we get the agent first to see if it exists.

File: pkg/invoke/invoker.go [hunk 2]
```diff
@@ -63,19 +70,28 @@ func (i *Invoker) getThread(ctx context.Context, agent *v2.Agent, input, threadN
 		}
 	}
 
+	workspaceID, err := i.workspaceClient.Create(ctx, i.workspaceProvider)
```
Comment (ibuildthecloud) [inline pkg/invoke/invoker.go]: this should be creating a workspace from the agent workspace, right?

File: pkg/storage/services/config.go [hunk 3]
```diff
@@ -15,7 +15,8 @@ type Config struct {
 	StorageToken      string `usage:"Token for storage access, will be generated if not passed"`
 	//AuditLogPath       string `usage:"Location of where to store audit logs"`
 	//AuditLogPolicyFile string `usage:"Location of audit log policy file"`
-	DSN string `usage:"Database dsn in driver://connection_string format" default:"sqlite://file:otto.db?_journal=WAL&cache=shared&_busy_timeout=30000"`
+	DSN               string `usage:"Database dsn in driver://connection_string format" default:"sqlite://file:otto.db?_journal=WAL&cache=shared&_busy_timeout=30000"`
+	WorkspaceProvider string `usage:"Workspace provider" default:"directory"`
```
Comment (ibuildthecloud) [inline pkg/storage/services/config.go]: No point in adding this config now as it really isn't usable.

File: pkg/invoke/invoker.go (hunk 2 above)
Comment (thedadams) [reply pkg/invoke/invoker.go]: Yes, fixed.

File: pkg/storage/services/config.go (hunk 3 above)
Comment (thedadams) [reply pkg/storage/services/config.go]: Hardcoded for now.

**Review by ibuildthecloud — APPROVED** [review]

**Pull request merged**